### PR TITLE
refactor: more optional chaining

### PR DESCRIPTION
**Pull request by johannschopplich**
## Describe the PR

Just a minor refactor to give you a taste for it. 😋

Some `typeof` checks are now skipped, which you might not prefer. Feel free to only adapt, what you like.
One day the Panel will be written in TS I guess, then it will get easier. 😇 

## Related issues/ideas

- Fixes #

## Ready?

- [ ] Unit tests for fixed bug/feature
- [ ] In-code documentation (wherever needed)
- [ ] CI checks pass

## When merging

- [ ] Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)
- [ ] Add changes to release notes draft in Notion

### CODE REVIEW

File: panel/src/components/Dialogs/Dialog.vue [hunk 1]
```diff
@@ -219,10 +219,7 @@ export default {
     focus() {
       if (this.$refs.dialog?.querySelector) {
         const btn = this.$refs.dialog.querySelector(".k-dialog-button-cancel");
-
-        if (typeof btn?.focus === "function") {
-          btn.focus();
-        }
+        btn?.focus?.();
```
Comment (afbora) [inline panel/src/components/Dialogs/Dialog.vue:222]: I'm not familiar with javascript but what if `btn.focus` is not a `function`? 🤔

Comment (johannschopplich) [reply panel/src/components/Dialogs/Dialog.vue:222]: That's why TypeScript is recommended for Kirby 4. 😜
The `querySelector` will return an `HTMLElement`, which has the focus method built-in: https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus

Comment (afbora) [reply panel/src/components/Dialogs/Dialog.vue:222]: OK for `querySelector`. What about other optional chaining changes where you made for function types?

Comment (johannschopplich) [reply panel/src/components/Dialogs/Dialog.vue:222]: If it's a `ref`, then it's a `HTMLElement` as well.

Comment (johannschopplich) [reply panel/src/components/Dialogs/Dialog.vue:222]: On the other elements where `selector[method]` applies, only `select`, `blur` and `focus` are the values used. Thus, we can skip the function check. If this method exists on the element, then it is intrinsically a function. Correct me if I'm wrong, of course. 🙂

Comment (distantnative) [reply panel/src/components/Dialogs/Dialog.vue:222]: Personally, I find `focus?.()` a bit weird. I know we need it in some places (e.g. I think in the overlay it's not clear what element we are calling this on), but in most of these places we should know what element we are referring/referencing and we should know if it has a focus method or not.

File: panel/src/components/Forms/Blocks/BlockSelector.vue [hunk 2]
```diff
@@ -114,10 +114,7 @@ export default {
     },
     navigate(index) {
       const ref = this.$refs["fieldset-" + index];
-
-      if (ref?.[0]) {
-        ref[0].focus();
-      }
+      ref?.[0].focus?.();
```
Comment (distantnative) [inline panel/src/components/Forms/Blocks/BlockSelector.vue]: Isn't this wrong? Question is not whether `ref` exists, but if it's empty or has actually a first element

Comment (johannschopplich) [reply panel/src/components/Forms/Blocks/BlockSelector.vue]: Absolutely! Great catch. Should be:

```js
ref?.[0]?.focus?.();
```

Comment (johannschopplich) [reply panel/src/components/Forms/Blocks/BlockSelector.vue]: As for other parts of the code: the optional call on `focus()` is probably redundant. However, as long as we don't use TypeScript we'd better be safe...

File: panel/src/components/Dialogs/Dialog.vue (hunk 1 above)
Comment (johannschopplich) [reply panel/src/components/Dialogs/Dialog.vue:222]: Weird it may be, but safer it is. 😉 TypeScript will help you from Kirby 4 onwards. 😇

Comment (distantnative) [reply panel/src/components/Dialogs/Dialog.vue:222]: As far as I can see, Typescript isn't really an option with Vue 2

Comment (johannschopplich) [reply panel/src/components/Dialogs/Dialog.vue:222]: Kirby 4 would be based on Vue 3, I guess... 👀

Comment (distantnative) [reply panel/src/components/Dialogs/Dialog.vue:222]: Yea, just thinking out loud here: I don't think we will ever be really able to do the big lift as from v2 to v3, rewriting everything for more than a year. So I'm still looking at ways to rather gradually introduce and move from Vue 2 to Vue 3 or from JS to TypeScript. But every time it feels more like they want us to pull the rug out beneath our feet.

Comment (johannschopplich) [reply panel/src/components/Dialogs/Dialog.vue:222]: Thanks for sharing your thoughts. Maybe rewriting using TS and Composition API won't be important and is optional. But one day, the jump to Vue 3 (without rewriting the entire panel, "only" the necessary parts) will probably make sense. Simpler plugins may just have to be recompiled with kirbyup (would then introduce a v3 option). More complex ones would probably still be tricky to migrate.

**Comment by nilshoerrmann** [comment]
Just a note from the outside: If you want to keep the core easily readable and understandable for others, it might be good not to use optional chaining everywhere you could. It raises complexity in some cases where an if statement might explain the intention of the code more directly.

**Comment by distantnative** [comment]
I moved the changes that I think we can add here: https://github.com/getkirby/kirby/pull/4133
Not removing the checks for functions for now.

**Pull request closed without merging**